python-sdk/tutorials/automl-with-azureml/forecasting-pipelines/auto-ml-forecasting-pipelines.ipynb

```diff
@@ -0,0 +1,783 @@
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "n_test_periods = 20\n",
+    "\n",
+    "test = pd.read_csv(\"oj-test.csv\", parse_dates=[time_column_name])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Upload data to datastore\n",
+    "The [Machine Learning service workspace](https://docs.microsoft.com/en-us/azure/machine-learning/service/concept-workspace), is paired with the storage account, which contains the default data store. We will use it to upload the train and test data and create [tabular datasets](https://docs.microsoft.com/en-us/python/api/azureml-core/azureml.data.tabulardataset?view=azure-ml-py) for training and testing. A tabular dataset defines a series of lazily-evaluated, immutable operations to load data from the data source into tabular representation."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from azureml.data.dataset_factory import TabularDatasetFactory\n",
+    "\n",
+    "datastore = ws.get_default_datastore()\n",
+    "train_dataset = TabularDatasetFactory.register_pandas_dataframe(\n",
+    "    train, target=(datastore, \"dataset/\"), name=\"dominicks_OJ_train\"\n",
+    ")\n",
+    "\n",
+    "test_dataset = TabularDatasetFactory.register_pandas_dataframe(\n",
+    "    test, target=(datastore, \"dataset/\"), name=\"dominicks_OJ_test\"\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Training"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Modeling\n",
+    "\n",
+    "For forecasting tasks, AutoML uses pre-processing and estimation steps that are specific to time-series. AutoML will undertake the following pre-processing steps:\n",
+    "* Detect time-series sample frequency (e.g. hourly, daily, weekly) and create new records for absent time points to make the series regular. A regular time series has a well-defined frequency and has a value at every sample point in a contiguous time span \n",
+    "* Impute missing values in the target (via forward-fill) and feature columns (using median column values) \n",
+    "* Create features based on time series identifiers to enable fixed effects across different series\n",
+    "* Create time-based features to assist in learning seasonal patterns\n",
+    "* Encode categorical variables to numeric quantities\n",
+    "\n",
+    "In this notebook, AutoML will train a single, regression-type model across **all** time-series in a given training set. This allows the model to generalize across related series. If you're looking for training multiple models for different time-series, please see the many-models notebook.\n",
+    "\n",
+    "You are almost ready to start an AutoML training job. First, we need to separate the target column from the rest of the DataFrame: "
```

Comment:
Should we say:
First, we need to define the target column.

Reply:
added